### PR TITLE
fix: Fix Api url path issue

### LINKED ISSUE #16377: [Bug]: Using expressions on the query editor causes the evaluated to behave incorrectly when the query is run
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Using a logical expression or a null coalease in the URL of an API causes the evaluation to incorrect pass values and return an error

### Steps To Reproduce

1. In an API query, pass query params as part of a conditional expression.
`https://mock-api.appsmith.com/{{Text1.text ? 'users' : 'user'}}` or https://mock-api.appsmith.com/{{Text1.text ?? 'users'}}
2. Run this query and observe that on running this query, the evaluated changes from the correct evaluation instead sending `{{Text1.text ` as the evaluated value as seen in the video

https://www.loom.com/share/59e7b899f5694fa8abcf9036ae8864f8

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by Irongade**
This PR introduces a fix for incorrect extraction of API url "path" fields when Dynamic bindings with expressions are inputed.

Example of those expressions are 

1. `https://mock-api.appsmith.com/{{Text1.text ? 'users' : 'user'}}`
2. `https://mock-api.appsmith.com/{{Text1.text ?? 'users'}`

For a typical API e.g.  www.google.com/users/api. the paths in this scenario corresponds to `"/users/api"`

When dynamic bindings with expressions (ternary expression for example) in them are written as Paths in the API url field e.g. `/{{Text1.text ? 'users' : 'user'}}`
The path was incorrectly being extracted. i.e. the path should correspond to this `/{{Text1.text ? 'users' : 'user'}}`, However, this was being extracted `/{{Text1.text `.

Fixes #16377 

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Jest test
- Cypress test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/cypress/support/Pages/ApiPage.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ export class ApiPage {
   private _confirmBeforeRunningAPI =
     "input[name='confirmBeforeExecute'][type='checkbox']";
   _saveAsDS = ".t--store-as-datasource";
+  _responseStatus = "//div[@id='root']";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/ApiPage.ts:43]: Here I think css selector ca be used. `_responseStatus = "div[id='root']";` Not an issue - but first preference goes to CSS since its faster comparatively & for complex ofcourse we choose xpath.

File: app/client/cypress/support/Pages/ApiPage.ts [hunk 2]
```diff
@@ -254,4 +255,9 @@ export class ApiPage {
       .should("be.visible")
       .click();
   }
+
+  ResponseStatusCheck(statusCode: string) {
+    cy.xpath(this._responseStatus).should("be.visible");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/ApiPage.ts:260]: Instead of direct check -`this.agHelper.AssertElementVisible(this._responseStatus);` can be used

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/BugTests/AbortAction_Spec.ts [hunk 3]
```diff
@@ -10,15 +15,14 @@ let dsName: any;
 const largeResponseApiUrl = "https://api.publicapis.org/entries";
 //"https://jsonplaceholder.typicode.com/photos";//Commenting since this is faster sometimes & case is failing
 
-export const ACTION_EXECUTION_CANCELLED = (actionName: string) =>
-  `${actionName} was cancelled`;
-
 describe("Abort Action Execution", function() {
   it("1. Bug #14006, #16093 - Cancel Request button should abort API action execution", function() {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/BugTests/AbortAction_Spec.ts:19]: Also - can we add one more case for this expression also - as given in PR desc:

`https://mock-api.appsmith.com/{{Text1.text ?? 'user1'}}`

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/BugTests/Bug16377_spec.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    apiPage.CreateAndFillApi(apiUrl, "BindingExpressions");
+    apiPage.RunAPI();
+    agHelper.AssertElementAbsence(
+      locator._specificToast(
+        createMessage(ERROR_ACTION_EXECUTE_FAIL, "BindingExpressions"),
+      ),
+    ); //Assert that an error is not returned.
+    apiPage.ResponseStatusCheck("200 OK");
+  });
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/BugTests/Bug16377_spec.ts:23]: Also - can we add one more case for this expression also - as given in PR desc:
`https://mock-api.appsmith.com/{{Text1.text ?? 'user1'}}`

Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/BugTests/Bug16377_spec.ts:23]: Also a negative case - when Text1.text does is not 'users' so we validate that url is `"url":"https://mock-api.appsmith.com/label"` and not `https://mock-api.appsmith.com/{{Text1.text` in negative case also
Screenshot 2022-09-08 at 1 57 10 PM

Comment (Irongade) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/BugTests/Bug16377_spec.ts:23]: This edge case has already been covered by existing tests. The culprit is the "?" operator. It doesn't if there are two or three.

Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/BugTests/Bug16377_spec.ts:23]: What if something changes tomorrow? You know its a single "?" causing trouble since you fixed it, but how is that an assurance that "??" will not break at all?

**Review by AmanAgarwal041 — APPROVED** [review]
